internal/monitors/disk/disk.go

```diff
@@ -0,0 +1,191 @@
+
+	// disk_octets.write
+	m.Output.SendDatapoint(datapoint.New("disk_octets.write", dimensions, datapoint.NewIntValue(int64(disk.WriteBytes)), datapoint.Counter, time.Time{}))
+
+	// disk_merged.read
+	m.Output.SendDatapoint(datapoint.New("disk_merged.read", dimensions, datapoint.NewIntValue(int64(disk.MergedReadCount)), datapoint.Counter, time.Time{}))
+
+	// disk_merged.write
+	m.Output.SendDatapoint(datapoint.New("disk_merged.write", dimensions, datapoint.NewIntValue(int64(disk.MergedWriteCount)), datapoint.Counter, time.Time{}))
+
+	// disk_time.read
+	m.Output.SendDatapoint(datapoint.New("disk_time.read", dimensions, datapoint.NewIntValue(int64(disk.ReadTime)), datapoint.Counter, time.Time{}))
+
+	// disk_time.write
+	m.Output.SendDatapoint(datapoint.New("disk_time.write", dimensions, datapoint.NewIntValue(int64(disk.WriteTime)), datapoint.Counter, time.Time{}))
+}
+
+func shouldSkipDisk(conf *Config, disk *gopsutil.IOCountersStat) (shouldSkip bool) {
+	_, match := conf.ignoreString[disk.Name]
+	match = match || utils.FindMatchString(disk.Name, conf.ignoreRegex)
+	if (match && *conf.IgnoreSelected) || (!match && !*conf.IgnoreSelected) {
+		shouldSkip = true
+	}
+	return
+}
+
+// EmitDatapoints emits a set of memory datapoints
+func (m *Monitor) emitDatapoints(conf *Config) {
+	iocounts, err := iOCounters()
+	if err != nil {
+		logger.WithError(err).Warningf("failed to load io counters. if this message repeates frequently there may be a problem")
+	}
+	// var total uint64
+	for key, disk := range iocounts {
+		// handle selecting disk
+		if shouldSkipDisk(conf, &disk) {
+			logger.Debugf("skipping disk '%s'", disk.Name)
+			continue
+		}
+
+		pluginInstance := strings.Replace(key, " ", "_", -1)
+		dimensions := map[string]string{"plugin": monitorType, "plugin_instance": pluginInstance}
+		if runtime.GOOS == "windows" {
+			m.processWindowsDatapoints(&disk, dimensions)
+		} else {
+			m.processLinuxDatapoints(&disk, dimensions)
+		}
+	}
+}
+
+// Configure is the main function of the monitor, it will report host metadata
+// on a varied interval
+func (m *Monitor) Configure(conf *Config) error {
+	if runtime.GOOS != "windows" {
+		logger.Warningf("'%s' monitor is in beta on this platform.  For production environments please use 'collectd/%s'.", monitorType, monitorType)
+	}
+
+	// create contexts for managing the the plugin loop
+	var ctx context.Context
+	ctx, m.cancel = context.WithCancel(context.Background())
+
+	// set env vars for gopsutil
+	if err := gopsutilhelper.SetEnvVars(map[string]string{gopsutilhelper.HostProc: conf.ProcFSPath}); err != nil {
+		return err
+	}
+
+	// convert array of strings and/or regexp to map of strings and array of regexp
+	var errs []error
+	conf.ignoreRegex, conf.ignoreString, errs = utils.RegexpStringsToRegexp(conf.Disks)
+	for _, err := range errs {
+		logger.Errorf(err.Error())
+	}
+
+	// default IgnoreSelected to true
+	if conf.IgnoreSelected == nil {
+		t := true
+		conf.IgnoreSelected = &t
```

Comment:
golib's `pointer.Bool(true)` is a bit cleaner.

Reply:
ack